### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by DamianX**
Implementation ~~stolen~~ largely based on TG.

The only test right now is supposed to check reagent recipe collisions, but it's wholesale copypasted from TG and is currently failing. I'll look into that, for the rest this should be ready for merge.

### CODE REVIEW

**Comment by DamianX** [comment]
Test should be passing as soon as #18337 is merged.

File: code/world.dm [hunk 1]
```diff
@@ -140,13 +140,12 @@ var/savefile/panicfile
 
 	src.update_status()
 
-	sleep_offline = 1
```
Comment (DamianX) [inline code/world.dm:143]: Just had a thought: should I only set this to 0 if we're running unit tests?

Comment (jknpj) [reply code/world.dm:143]: 🤔

Comment (Exxion) [reply code/world.dm:143]: Probably, but it is unlikely to ever have a functional effect

Comment (DamianX) [reply code/world.dm:143]: I'll just leave this as is, then

**Pull request merged**